### PR TITLE
Issue #3657: skip unnecessary exception 'Severity not set, ignoring e…

### PR DESCRIPTION
**Pull request by romani**
Issue #3657

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/ConfigurationLoader.java [hunk 1]
```diff
@@ -495,5 +496,18 @@ public void endElement(String uri,
...
             }
         }
+
+        /**
+         * Util method to recheck attribute in module.
+         * @param module modeul to check
+         * @param attributeName name of atteribute in module to find
+         * @return true if attribute is present in module
+         */
+        private boolean containsAttribute(Configuration module, String attributeName) {
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/ConfigurationLoader.java]: To me, I think it makes more sense for this to be a new method in `Configuration`.

File: src/main/java/com/puppycrawl/tools/checkstyle/ConfigurationLoader.java [hunk 2]
```diff
@@ -495,5 +496,18 @@ public void endElement(String uri,
                 }
             }
         }
+
+        /**
+         * Util method to recheck attribute in module.
+         * @param module modeul to check
+         * @param attributeName name of atteribute in module to find
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/ConfigurationLoader.java]: atteribute -> attribute

File: src/main/java/com/puppycrawl/tools/checkstyle/ConfigurationLoader.java [hunk 3]
```diff
@@ -495,5 +496,18 @@ public void endElement(String uri,
                 }
             }
         }
+
+        /**
+         * Util method to recheck attribute in module.
+         * @param module modeul to check
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/ConfigurationLoader.java]: modeul -> module

Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/ConfigurationLoader.java]: done

File: src/main/java/com/puppycrawl/tools/checkstyle/ConfigurationLoader.java (hunk 2 above)
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/ConfigurationLoader.java]: done

File: src/main/java/com/puppycrawl/tools/checkstyle/ConfigurationLoader.java (hunk 1 above)
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/ConfigurationLoader.java]: yes, but I do not want to extend API for now. So it is hosted only in one implementation that need it. Whole design is interesting, we have API interface, but code is not ready to have any other from DefaultConfiguration. This need to be reconsidered completely in checkstyle8.

So, for now, no extending for api.

**Comment by romani** [comment]
Investigation:

Code changes for Tracing output:
$ git diff
diff --git a/src/test/java/com/puppycrawl/tools/checkstyle/ConfigurationLoaderTest.java b/src/test/java/com/puppycrawl/tools/checkstyle/ConfigurationLoaderTest.java
index af5fa23..9eb11f5 100644
--- a/src/test/java/com/puppycrawl/tools/checkstyle/ConfigurationLoaderTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/ConfigurationLoaderTest.java
@@ -388,6 +388,8 @@ public class ConfigurationLoaderTest {
             param[3] = Attributes.class;
             final Method method = aClass.getDeclaredMethod("startElement", param);
             final Object objParent = ctorParent.newInstance(null, true);
+            System.out.println("[MYOUT] contructor - " + constructor);
+            System.out.println("[MYOUT] objParent - " + objParent);
             final Object obj = constructor.newInstance(objParent); // <----- HERE IS EXCEPTION
 
             method.invoke(obj, "", "", "hello", null);
```

Commands:
```
mvn nondex:nondex >nondex.log
...
mvn clean test >test.log
....
```

Out put diff:
```
$ grep "\[MYOUT\]" test.log 
[MYOUT] contructor - public com.puppycrawl.tools.checkstyle.ConfigurationLoader$InternalLoader(com.puppycrawl.tools.checkstyle.ConfigurationLoader) throws org.xml.sax.SAXException,javax.xml.parsers.ParserConfigurationException
[MYOUT] objParent - com.puppycrawl.tools.checkstyle.ConfigurationLoader@45eb6b7d

$ grep "\[MYOUT\]" nondex.log 
[MYOUT] contructor - public com.puppycrawl.tools.checkstyle.ConfigurationLoader$InternalLoader(com.puppycrawl.tools.checkstyle.ConfigurationLoader) throws org.xml.sax.SAXException,javax.xml.parsers.ParserConfigurationException
[MYOUT] objParent - com.puppycrawl.tools.checkstyle.ConfigurationLoader@45eb6b7d
[MYOUT] contructor - public com.puppycrawl.tools.checkstyle.ConfigurationLoader$InternalLoader(org.powermock.core.IndicateReloadClass)
[MYOUT] objParent - com.puppycrawl.tools.checkstyle.ConfigurationLoader@aa9bf75
[MYOUT] contructor - public com.puppycrawl.tools.checkstyle.ConfigurationLoader$InternalLoader(org.powermock.core.IndicateReloadClass)
[MYOUT] objParent - com.puppycrawl.tools.checkstyle.ConfigurationLoader@3f793327
[MYOUT] contructor - public com.puppycrawl.tools.checkstyle.ConfigurationLoader$InternalLoader(org.powermock.core.IndicateReloadClass)
[MYOUT] objParent - com.puppycrawl.tools.checkstyle.ConfigurationLoader@671f8e4a
```

**Comment by romani** [comment]
squashed and rebased and merged manually.

**Pull request closed without merging**